cmd/bot/run.go

```diff
@@ -100,6 +115,11 @@ func ProcessUnprovenBotDelegatedWithdrawals(ctx context.Context, log log.Logger,
 	}
 
 	for _, unproven := range unprovens {
+		// In order to avoid re-processing the same withdrawal, we use a tombstone to mark the withdrawal as processed.
+		if hasWithdrawalRecentlyProcessed(&unproven, tombstone) {
```

Comment:
If we already have `isPendingAndChainNonceEqual` check before calling `ProcessUnprovenBotDelegatedWithdrawals`, this check should be redundant.

Since there is no pending txns, and this loop will only send at most once for the same ID.
Without this check, the event will get the error `OptimismPortal: withdrawal hash has already been proven` in dry run, and marked proven then.
But with this check, we have to wait at least 10 min before marking it as proven.

What we want to ensure is that there are no pending txns which handle the same ID. But this check can't achieve this goal.

Reply:
I agree with your analysis. At first, I only implemented the nonce check, but during testing, I found that re-processing still existed. I have no clue that why re-processing still happens even with nonce check, just suspect it’s related to the L1 node(?)

I didn't spend time to troubleshoot this issue, just only add the local tombstone check to check more strictly.

Reply:
Hi @owen-reorg, thanks for your review and sorry for late reply.

I have removed `hasWithdrawalRecentlyProcessed` check and added `PendingTxnCheck`check at c18eb3c052d4ef6f6a24e547ac24d5aff8e31d86

Thanks again.